File: 2015/day1.js
const fs = require('fs')
const input = fs.readFileSync(`./input/day1.txt`, 'utf8').trim()

const floorFinder = input => input.split('').reduce((total, char) => total + (char === '(' ? 1 : -1), 0)
console.log(floorFinder(input))


const basementFinder = input => {
  const inputs = input.split('')
  let floor = 0, i = 0
  while (floor > -1) {
    inputs[i] === '(' ? floor++ : floor--
    i++
  }
  return i
}
console.log(basementFinder(input))

File: 2015/day1.rb
input = File.read('./input/day1.txt')

up, down = input.strip.split('').partition{ |c| c == '(' }.map(&:length)
puts up - down

def findBasement(input)
  floor = 0
  input.strip.split('').each_with_index do |c, i|
    floor += c == '(' ? 1 : -1
    return i + 1 if floor < 0
  end
end

puts findBasement input
